src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -41,12 +43,10 @@ public DiscountResponseVO create(DiscountRequestVO request) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public DiscountResponseVO update(Long id, DiscountRequestVO request) {
-        Discount discount = discountRepository.findById(id).orElse(null);
-        if (discount != null) {
-            discountMapper.updateEntity(request, discount);
-            return discountMapper.toVO(discountRepository.save(discount));
-        }
-        return null;
+        Discount discount = discountRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Could not find discount with id: " + id));
+        discountMapper.updateEntity(request, discount);
+        return discountMapper.toVO(discountRepository.save(discount));
```

Comment:
the same here
plese split this into two lines

Reply:
done